src/cli/commands/add.js

```diff
@@ -204,7 +204,7 @@ export async function run(
     throw new MessageError(reporter.lang('missingAddDependencies'));
   }
 
-  const lockfile = await Lockfile.fromDirectory(config.cwd, reporter);
+  const lockfile = await Lockfile.fromDirectory(config, config.cwd, reporter);
```

Comment:
Not really, but it makes sense semantically to specify the path, otherwise it should be renamed to `fromConfig` - and as you mentioned in the comment below, it would probably be for the best to keep this function not too tightly coupled with the config object.